packages/Algernon.package/ALGCore.class/instance/deactivate..st

```diff
@@ -3,6 +3,7 @@ deactivate: hand
 	self active
 		ifFalse: [^ self].
 	self lastInputString: self topmostQuery.
+	self lastWorldBounds: World bounds corner.
```

Comment:
You need to insert a checkWorldBounds before updating it. Because when I resize the window while Algernon is opened and close Algernon after the resize the new resolution is saved, therefore checkWorldBounds doesn't do anything on the next activation; but Algernon could still be out of view.

Reply:
I think our tutor even said that its okay to have the repositioning only affect Algernon after closing and reopening, so it is a valid process as far as I know
Or maybe I do not understand your point very well

Reply:
Yes that's correct, but when the resizing is done while Algernon is open (meaning after the activation and call of checkWorldBounds and before the deactivation and the setting of lastWorldBounds) your code saves the new window size on closing Algernon, and when Algernon is activated again checkWorldBounds does nothing, even trough Algernon could be out of view.

To try it for your self, just open Algernon, drag it on the right edge, keep it open and make the squeak window smaller, when you now reopen Algernon it will be still out of view.

Reply:
That issue would probably also be resolved by just checking if Algernon is in view, as suggested in my other comment.